app/src/main/java/com/pvp/app/ui/screen/calendar/MonthlyCalendarScreen.kt

```diff
@@ -0,0 +1,247 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
+import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.pvp.app.model.Task
+import com.pvp.app.ui.screen.calendar.CalendarUiState
+import com.pvp.app.ui.screen.calendar.DateUtil
+import com.pvp.app.ui.screen.calendar.Day
+import com.pvp.app.ui.screen.calendar.MonthlyCalendarViewModel
+import com.pvp.app.ui.screen.calendar.getDisplayName
+import java.time.LocalDate
+import java.time.YearMonth
+
+@Composable
+fun MonthlyCalendarScreen(
+    model: MonthlyCalendarViewModel = hiltViewModel(),
+) {
+    val state by model.state.collectAsState()
+    var showDialog by remember { mutableStateOf(false) }
+    var selectedDate by remember { mutableStateOf(LocalDate.now()) }
+    var dateTasks by remember { mutableStateOf<List<Task>>(emptyList()) }
+
+    MonthlyCalendar(
+        days = DateUtil.daysOfWeek,
+        month = state.yearMonth,
+        dates = state.dates,
+        onClickListener = { date ->
+            showDialog = true
+            selectedDate = date.date
+            dateTasks = date.tasks
+        }
+    )
+
+    if (showDialog) {
+        DayDialog(
+            date = selectedDate,
+            tasks = dateTasks,
+            onDismissRequest = { showDialog = false }
+        )
+    }
+}
+
+@Composable
+fun DayDialog(
+    date: LocalDate,
+    tasks: List<Task> = listOf(),
+    onDismissRequest: () -> Unit
+) {
+    Dialog(
+        onDismissRequest = { onDismissRequest() },
+        properties = DialogProperties(usePlatformDefaultWidth = false)
+    ) {
+        Day(
+            name = date.toString(),
+            tasks = tasks
+        )
+    }
+}
+
+@Composable
+fun MonthlyCalendar(
+    days: Array<String>,
+    month: YearMonth,
+    dates: List<CalendarUiState.DateEntry>,
+    onClickListener: (CalendarUiState.DateEntry) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+    ) {
+        Row {
+            repeat(days.size) {
+                val item = days[it]
```

Comment:
You could leave it as it is, but still, writing some ideas for this as I noticed this happens a lot. This is just my opinion.

This is unnecessary and could be called within parameter assignation `day = days[it]`.

Now variable is created O(n) times and only to be referenced in a single spot. This also doesn't change readability for the better, since a very simple action is being made. In a very insignificant way it actually makes the readability worse, more stuff happens by assigning the resolved value.

If the action was longer (like 100+ characters long) or complex (made of multiple steps), then it would make sense to separate these values.